### PR TITLE
target/ppc: implement vgnb

### PR DESCRIPTION
**Pull request by mferst**

### CODE REVIEW

**Review by lffpires — APPROVED** [review]
LGTM! 2 **very** minor comments.

File: target/ppc/insn32.decode [hunk 1]
```diff
@@ -53,6 +53,9 @@
 &VX             vrt vra vrb
 @VX             ...... vrt:5 vra:5 vrb:5 .......... .   &VX
 
+&VX_n           rt vrb n
+@VX_n           ...... rt:5 -- n:3 vrb:5 ...........    &VX_n
```
[the remark this answers is not in the document]
Comment (mferst) [reply target/ppc/insn32.decode:57]: Fixed in https://github.com/PPC64/qemu/pull/49/commits/8530506a48a0e8ce48e6f772a560e91f37c98193

File: target/ppc/translate/vmx-impl.c.inc [hunk 2]
```diff
@@ -1228,6 +1228,45 @@ GEN_VXFORM_DUAL(vsplth, PPC_ALTIVEC, PPC_NONE,
...
+static bool trans_VGNB(DisasContext *ctx, arg_VX_n *a)
+{
+    TCGv_i64 vrb, tmp, rt;
+    int in = 63, out = 63;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VECTOR(ctx);
+
+    if (a->n < 2) {
+        return true;
```
[the remark this answers is not in the document]
Comment (mferst) [reply target/ppc/translate/vmx-impl.c.inc:1245]: Fixed in https://github.com/PPC64/qemu/pull/49/commits/8530506a48a0e8ce48e6f772a560e91f37c98193

File: target/ppc/insn32.decode [hunk 3]
```diff
@@ -367,7 +367,7 @@ DSCRIQ          111111 ..... ..... ...... 001100010 .   @Z22_tap_sh_rc
 
 ## Vector Bit Manipulation Instruction
 
-VGNB            000100 ..... -- ... ..... 10011001100   @VX_n
+VGNB            000100 ..... .. ... ..... 10011001100   @VX_n
```
Comment (lffpires) [inline target/ppc/insn32.decode:370]: This change (replacing '-'s with '.'s) was supposed to be done in the format definition (VX_n), not here.

Comment (mferst) [reply target/ppc/insn32.decode]: Oops. Fixed in https://github.com/PPC64/qemu/pull/49/commits/7b4fa496310615bf5f7eda7b950fb96a186ff07b

**Review by lffpires — APPROVED** [review]

**Review by vcoracolombo — APPROVED** [review]

**Comment by mferst** [comment]
Merged in upstream: https://lists.gnu.org/archive/html/qemu-ppc/2022-03/msg00002.html

**Pull request closed without merging**